### PR TITLE
Elevator Iteration 1

### PR DESCRIPTION
**Pull request by cville989**

### CODE REVIEW

File: Elevator.java [hunk 1]
```diff
@@ -1 +1,96 @@
...
+                return null;
+            }
+        }
+        
+        ArrayList<Floor.Rider> items = contents;
+        contents = null;
+        empty = true;
+        notifyAll();
+        return items;
+    }
```
Comment (cville989) [inline Elevator.java:77]: Indentation

File: Elevator.java [hunk 2]
```diff
@@ -1 +1,96 @@
...
+	            try {
+	                wait();
+	            } catch (InterruptedException e) {
+	                return;
+	            }
+	        }
+	        contents = items;
+	        empty = false;
+	        notifyAll();
+	    }
```
Comment (cville989) [inline Elevator.java:59]: Indentation

File: Elevator.java [hunk 3]
```diff
@@ -1 +1,96 @@
...
+			System.out.print("Doors Open");
+		}
+		else{
+			System.out.print("Doors Closed");
+			while(curFloor != floorToGo) {
+				curFloor--;
+			}
+			System.out.print("Doors Open");
+		}
+	}
```
Comment (cville989) [inline Elevator.java:46]: These will need to include the timing delays we found in iteration 0.

File: Elevator.java [hunk 4]
```diff
@@ -1 +1,96 @@
...
+	
+	public Elevator(int floorToGo, int curFloor) {
+		this.floorToGo = floorToGo;
+		this.curFloor = curFloor;
+	}
+	
+	public void up() {
+		
+		if (curFloor == maxFloor && floorToGo == maxFloor) {
+			System.out.print("Doors Open");
```
Comment (cville989) [inline Elevator.java:23]: Not a comment to fix now, but when we have the scheduler made, we'll have to define the information being passed back to the Floors

File: Elevator.java [hunk 5]
```diff
@@ -1 +1,96 @@
...
+	
+	public int getMaxFloor() {
+		return maxFloor;
+	}
+	
+	public int getFloorToGo() {
+		return floorToGo;
+	}
+
+}
```
Comment (cville989) [inline Elevator.java:96]: Should there be a method for Passengers selecting their destination?

File: Elevator.java [hunk 6]
```diff
@@ -1 +1,96 @@
...
 
+import java.util.ArrayList;
+
+
+public class Elevator {
+	
+	private int maxFloor = 2;
+	private int groundFloor = 0;
+	private int curFloor;
+	private int floorToGo;
```
Comment (cville989) [inline Elevator.java:11]: The elevator doesn't really have a destination, the passengers have destination floors. Maybe we should create a Passenger object for this purpose

File: Elevator.java [hunk 7]
```diff
@@ -1 +1,96 @@
...
+			}
+			System.out.print("Doors Open");
+		}
+	}
+	
+	public void down() {
+		
+		if (curFloor == groundFloor && floorToGo == groundFloor) {
+			System.out.print("Doors Open");
+		}
```
Comment (cville989) [inline Elevator.java:38]: Don't think you need to check for it to be the ground floor, if you make it curFloor == floorToGo, that'd fix 2 edge cases. Same with maxFloor in the up method.

File: Elevator.java (hunk 6 above)
Comment (josh16noronha) [reply Elevator.java:11]: We don't need a Passenger class the elevator acts as the Passenger.

File: Elevator.java [hunk 8]
```diff
@@ -5,47 +5,80 @@
...
 	private int floorToGo;
-	private ArrayList<Floor.Rider> contents = null; // contents
-	private boolean empty = true; // empty?
+	//private ArrayList<Floor.Rider> contents = null; // contents
+	//private boolean empty = true; // empty?
 	
-	public Elevator(int floorToGo, int curFloor) {
+	public Elevator(int floorToGo, int curFloor, int maxFloor, int groundFloor) {
 		this.floorToGo = floorToGo;
 		this.curFloor = curFloor;
```
Comment (cville989) [inline Elevator.java:17]: These should not be defined in the constructor, there should be a separate method to assign a destination, and there should be a default starting floor

File: Elevator/Elevator.java [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+	private int floorToGo;
+	private Scheduler schedule;
+
+	
+	public Elevator(int floorToGo, int curFloor, int maxFloor, int groundFloor) {
+		this.floorToGo = floorToGo;
+		this.curFloor = curFloor;
+		this.maxFloor = maxFloor;
+		this.groundFloor = groundFloor;
+	}
```
Comment (cville989) [inline Elevator/Elevator.java:21]: Should not pass current floor or destination via the constructor, set the current floor by default to something, maybe the ground floor, the elevator shouldn't need a specific destination floor.

File: Elevator/Elevator.java [hunk 10]
```diff
@@ -0,0 +1,197 @@
...
+	 */
+	public void pressButton() throws InterruptedException {
+		if (floorToGo < curFloor) {
+			down();
+		}
+		else {
+			up();
+		}
+		
+	}
```
Comment (cville989) [inline Elevator/Elevator.java:78]: Not sure if this is necessary

File: Elevator/Elevator.java [hunk 11]
```diff
@@ -0,0 +1,197 @@
...
+	 * per the passenger's request. 
+	 * 
+	 * @return true or false If the request array is null or not
+	 * @throws InterruptedException 
+	 */
+	private boolean elevatorActivated() throws InterruptedException {
+		Optional<ArrayList<String>> obj = schedule.getRequest(curFloor);
+		ArrayList <String> arr = obj.get();
+		
+		if(obj.isEmpty()) {
```
Comment (cville989) [inline Elevator/Elevator.java:92]: This should be !obj.isEmpty()
Right now it only does anything if nothing is in the optional

File: Elevator/Elevator.java (hunk 10 above)
Comment (josh16noronha) [reply Elevator/Elevator.java:78]: I will leave this for future iterations.

File: Elevator/Elevator.java (hunk 9 above)
Comment (josh16noronha) [reply Elevator/Elevator.java:21]: Issue Fixed.

File: Elevator/Elevator.java (hunk 11 above)
Comment (josh16noronha) [reply Elevator/Elevator.java:92]: Issue Fixed.

**Review by Keeferf — APPROVED** [review]
good to merge

**Pull request merged**